File: src/com/leetcode/Medium/BoatsToSavePeople.java
package com.leetcode.Medium;

import java.util.Arrays;

public class BoatsToSavePeople {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);

        int minNumberOfBoats = 0;
        int i = 0, j = people.length - 1;
        while (i <= j) {
            if (people[i] + people[j] <= limit)
                i++;

            j--;
            minNumberOfBoats++;
        }

        return minNumberOfBoats;
    }
}

File: JavaScript/Medium/BoatsToSavePeople.js
/**
 * 881. Boats to Save People
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function (people, limit) {
  people.sort((a, b) => a - b);

  let minNumberOfBoats = 0;
  let i = 0, j = people.length - 1;
  while (i <= j) {
    if (people[i] + people[j] <= limit) {
      i++;
    }

    j--;
    minNumberOfBoats++;
  }

  return minNumberOfBoats;
};
